client/src/main/java/keywhiz/client/KeywhizClient.java

```diff
@@ -350,6 +350,14 @@ public SanitizedSecret getSanitizedSecretByName(String name) throws IOException
     return mapper.readValue(response, SanitizedSecret.class);
   }
 
+  public List<SanitizedSecret> getDeletedSecretsByName(String name) throws IOException {
+    checkArgument(!name.isEmpty());
+    String response =
+        httpGet(baseUrl.resolve(format("/admin/secrets/deleted/%s", name)).newBuilder().build());
+    return mapper.readValue(response, new TypeReference<>() {
+    });
+  }
```

Comment:
This class mostly provides numerous bad examples, but we should opt for the HttpUrl.builder()-based approach (which a couple of the methods in the class do use) for building up the URL as opposed to String.format(), which does not perform any of the necessary escaping.

Reply:
I think I get what you're saying here. this is what I'm changing it to:

```
String response =
        httpGet(baseUrl.newBuilder()
            .addPathSegment("admin")
            .addPathSegment("secrets")
            .addPathSegment("deleted")
            .addPathSegment(name)
            .build());
```